runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -1918,6 +1926,26 @@ private Optional<WorkerMessage> createWorkerMessageForPerWorkerMetrics() {
     return Optional.of(workUnitClient.createWorkerMessageFromPerWorkerMetrics(perWorkerMetrics));
   }
 
+  private void readAndSaveWorkerMessageResponseForStreamingScalingReportResponse(
+      List<WorkerMessageResponse> responses) {
+    for (WorkerMessageResponse response : responses) {
+      if (response.getStreamingScalingReportResponse() != null) {
```

Comment:
would it be possible to get multiple responses? If so perhaps it would be best to find the last value in the loop and then just set once instead of possibly setting multiple times.